pkg/platform/runtime/setup/setup.go

```diff
@@ -341,16 +328,21 @@ func (s *Setup) setupArtifactSubmitFunction(a artifact.ArtifactDownload, buildRe
 			return
 		}
 
-		if err := s.setupArtifact(buildResult.BuildEngine, a.ArtifactID, a.UnsignedURI); err != nil {
-			if err != nil {
-				name := setup.ResolveArtifactName(a.ArtifactID)
-				errors <- locale.WrapError(err, "artifact_setup_failed", "", name, a.ArtifactID.String())
-			}
+		as, err := s.selectArtifactSetupImplementation(setup.BuildEngine(), a.ArtifactID)
+		if err != nil {
+			errors <- errs.Wrap(err, "Failed to select artifact setup implementation")
+			return
+		}
```

Comment:
Reading through the changes I'm not seeing why it was necessary to extract this behaviour from `setupArtifact`. Can we revert this?

Reply:
Probably leftover from a previous attempt. I'll revert it.